src/codegen.jl

```diff
@@ -355,20 +339,33 @@ end
 
 function debug_actions(machine::Machine)
     actions = Set{Symbol}()
-    for trans in values(machine.transitions)
-        for (_, as) in values(trans)
-            union!(actions, as)
+    for s in traverse(machine.start)
+        for (e, t) in s.edges
+            union!(actions, a.name for a in e.actions)
         end
     end
     for as in values(machine.eof_actions)
-        union!(actions, as)
+        union!(actions, a.name for a in as)
     end
     function log_expr(name)
         return :(push!(logger, $(QuoteNode(name))))
     end
     return Dict(name => log_expr(name) for name in actions)
 end
 
+# Generic foldr.
+function foldr(op::Function, x0, xs)
```

Comment:
It is plausible, but this works in the same way as `Base.foldr` in 99.9% of cases and I am the only developer of this package at the moment. So, I have no motivation to rename it to another name. Perhaps the best way is to send a patch to the Julia upstream.